sdk/digitaltwins/mgmt-v2020_12_01/src/test/java/com/microsoft/azure/management/digitaltwins/v2020_12_01/DigitalTwinsLifecycleTests.java

```diff
@@ -0,0 +1,132 @@
+import com.microsoft.rest.RestClient;
+import com.microsoft.azure.management.resources.implementation.ResourceManager;
+import org.junit.Assert;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class DigitalTwinsLifecycleTests extends TestBase {
+
+    protected ResourceManager resourceManager;
+    protected DigitalTwinsManager digitalTwinsManager;
+    private static String rgName;
+    protected String domain;
+    private static String defaultInstanceName = "DigitalTwinsSdk";
+    private static String defaultRegion = Region.US_WEST_CENTRAL.toString();
+    private static String defaultResourceGroupName = "rg2b9842374ecf6";
+
+    @Override
+    protected void initializeClients(RestClient restClient, String defaultSubscription, String domain) throws IOException {
+        resourceManager = ResourceManager
+            .authenticate(restClient)
+            .withSubscription(defaultSubscription);
+
+        digitalTwinsManager = DigitalTwinsManager
+            .authenticate(restClient, defaultSubscription);
+
+        this.domain = domain;
+    }
+
+    @Test
+    public void lifecycleTest() throws InterruptedException {
+        //rgName = SdkContext.randomResourceName("rg", 20);
+        rgName = defaultResourceGroupName;
+
+        ResourceGroup group = resourceManager.resourceGroups()
+            .define(rgName)
+            .withRegion(defaultRegion)
+            .create();
+
+        Assert.assertNotNull(group);
+
+        try {
+            CheckNameResultInner checkNameResult = digitalTwinsManager.inner().digitalTwins()
+                .checkNameAvailability(defaultRegion, defaultInstanceName);
+
+            if (!checkNameResult.nameAvailable()) {
+                Iterator<DigitalTwinsDescription> allDigitalTwins = digitalTwinsManager.digitalTwins().listAsync().toBlocking().getIterator();
+
+                while (allDigitalTwins.hasNext()) {
+                    DigitalTwinsDescription digitalTwin = allDigitalTwins.next();
+                    if (digitalTwin.name().equals(defaultInstanceName)){
+                        String existingResourceGroupName = digitalTwin.resourceGroupName();
+                        digitalTwinsManager.inner().digitalTwins().delete(existingResourceGroupName, defaultInstanceName);
+                        break;
+                    }
+                }
+
+                checkNameResult = digitalTwinsManager.inner()
+                    .digitalTwins()
+                    .checkNameAvailability(defaultRegion, defaultInstanceName);
+
+                Assert.assertTrue(checkNameResult.nameAvailable());
+            }
+
+            // Create DigitalTwins resource
+            DigitalTwinsDescription instance = digitalTwinsManager
+                .digitalTwins()
+                .defineDigitalTwinsInstance(defaultInstanceName)
+                .withRegion(defaultRegion).withExistingResourceGroup(rgName)
```

Comment:
Shouldn't these be on separate lines? That is `.withExistingResourceGroup` deserves to be a top-level indent like the others, right?